doc/cookbook/custom_dna4.cpp

```diff
@@ -0,0 +1,57 @@
+#include <seqan3/alphabet/nucleotide/nucleotide_base.hpp>
+
+class my_dna4 : public seqan3::nucleotide_base<my_dna4, 4>
+{
+public:
+    using nucleotide_base<my_dna4, 4>::nucleotide_base; // Pull constructors of the base class into public scope
+
+private:
+    // Value to char conversion table.
+    static constexpr char_type rank_to_char[alphabet_size] {'A', 'C', 'G', 'T'}; // rank 0,1,2,3
+
+    // Char-to-value conversion table. This is where char conversion for assign_char() is handled!
+    static constexpr std::array<rank_type, 256> char_to_rank
+    {
+        [] () constexpr
+        {
+            std::array<rank_type, 256> conversion_array{}; // by default now, everything has rank 0 which equals `A`
+
+            conversion_array['C'] = conversion_array['c'] = 1;
+            conversion_array['G'] = conversion_array['g'] = 2;
+            conversion_array['T'] = conversion_array['t'] = 3;
+            conversion_array['U'] = conversion_array['T']; // set U equal to T
+            conversion_array['u'] = conversion_array['t']; // set u equal to t
+
+            return conversion_array;
+        }()
+    };
+
+    // The forward declaration of the complement table. With this, my_dna4 models seqan3::nucleotide_alphabet.
+    static const std::array<my_dna4, alphabet_size> complement_table;
+
+    friend nucleotide_base<my_dna4, 4>; // Grant seqan3::nucleotide_base access to private/protected members.
+    friend nucleotide_base<my_dna4, 4>::base_t; // Grant seqan3::alphabet_base access to private/protected members.
+};
+
+constexpr my_dna4 operator""_my_dna4(char const c) noexcept
+{
+    return my_dna4{}.assign_char(c);
+}
+
+// The definition of the complement table. With this, my_dna4 models seqan3::nucleotide_alphabet.
+constexpr std::array<my_dna4, my_dna4::alphabet_size> my_dna4::complement_table
+{
+    'T'_my_dna4,    // the complement of 'A'_my_dna4
+    'G'_my_dna4,    // the complement of 'C'_my_dna4
+    'C'_my_dna4,    // the complement of 'G'_my_dna4
+    'A'_my_dna4     // the complement of 'T'_my_dna4
+};
+
+int main()
+{
+    my_dna4 my_letter{'C'_my_dna4};
+
+    my_letter.assign_char('S'); // characters other than A,C,T,G are implicitly converted to A.
+    if (my_letter.to_char() == 'A')
+         "yeah\n"; // "yeah";
```

Comment:
This creates the char array and then destroys it?

Reply:
```suggestion
         std::cout << "yeah\n"; // "yeah";
```